From b2bfe49c875e487f5af76ce2679fbfd1c58440ff Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Fri, 18 Aug 2017 17:24:04 +0800
Subject: [PATCH 2488/5242] MLK-16221-2 usb: cdns3: gadget: avoid visiting
 register when detach

commit  e9c6c62592888482f25622c2274d29378df44f8c from
https://source.codeaurora.org/external/imx/linux-imx.git

When the port does not connect to host, the controller's gadget
mode is enabled, so we need to avoid visiting register at this situation.

Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/cdns3/gadget.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/cdns3/gadget.c b/drivers/usb/cdns3/gadget.c
index 09416c3..921a270 100644
--- a/drivers/usb/cdns3/gadget.c
+++ b/drivers/usb/cdns3/gadget.c
@@ -1839,7 +1839,8 @@ static int usb_ss_gadget_udc_stop(struct usb_gadget *gadget)
 	spin_lock_irqsave(&usb_ss->lock, flags);
 	usb_ss->gadget_driver = NULL;
 	/* disable interrupt for device */
-	gadget_writel(usb_ss, &usb_ss->regs->usb_ien, 0);
+	if (usb_ss->start_gadget)
+		gadget_writel(usb_ss, &usb_ss->regs->usb_ien, 0);
 	spin_unlock_irqrestore(&usb_ss->lock, flags);
 
 	return 0;
-- 
1.7.9.5

